scripts/export_requirements.py

```diff
@@ -55,7 +57,86 @@ def get_hash(content: dict) -> str:
     return hash == get_hash(content)
 
 
-def main(req_path: os.PathLike, should_validate_hash: bool = True) -> typing.Optional[int]:
+def _extract_packages_from_requirements(requirements: str) -> list:
+    """Extract a list of packages from the provided requirements str."""
+    req = requirements.split("\n")
+    packages = set()
+    for line in req.copy():
+        if line.startswith("#"):
+            continue
+        if not len(line.strip()):
+            continue
+
+        # requirement files we will be parsing can have `;` or =<>~
+        match = PACKAGE_REGEX.match(line)
+        if match is None:
+            continue
+
+        package = match.group("package").strip().lower().replace("_", "-")
```

Comment:
Why do we do `.replace("_", "-")`? I don't see any packages using `_`.

Reply:
Pypi replaces all `-` with `_` (or vice versa, they're the same character) and while poetry is supposed to normalize them to the lockfile, I don't want to come across a weird poetry bug that doesn't

Reply:
Hmm, I think it would be better to let poetry handle it and in case we come across such a bug in future we could add this edit functionality again.

In case, you still want that to stay, you should document why you do that in the code.